fibsem/microscope.py

```diff
@@ -2866,6 +2883,61 @@ def draw_bitmap_pattern(self, pattern_settings: FibsemBitmapSettings):
         self._patterns.append(pattern)
         return pattern
 
+    def _resize_bitmap_to_pattern(
+        self, pattern_settings: FibsemBitmapSettings
+    ) -> NDArray[np.float_ | np.uint8]:
+        points = pattern_settings.bitmap
+
+        if points is None:
+            raise ValueError(
+                "Unable to resize bitmap as FibsemBitmapSettings.bitmap is None"
+            )
+
+        # Get pitch to calculate expected pixel size
+        rectangle = self.connection.patterning.create_rectangle(
+            center_x=pattern_settings.centre_x,
+            center_y=pattern_settings.centre_y,
+            width=pattern_settings.width,
+            height=pattern_settings.height,
+            depth=pattern_settings.depth,
+        )
+
+        new_shape = (
+            int(round(pattern_settings.height / rectangle.pitch_y)),
+            int(round(pattern_settings.width / rectangle.pitch_x)),
+        )
+
+        # Disable after calculations just in case values are cleared
+        rectangle.enabled = False
```

Comment:
Not sure if it's possible in the current version, but it would be better to remove the pattern. Just a note